app/ide-desktop/lib/dashboard/src/authentication/src/authentication/service.tsx

```diff
@@ -34,13 +40,42 @@ const AMPLIFY_CONFIGS = {
   },
 };
 
+// ==========================
+// === Authentication API ===
+// ==========================
+
+/** `window.authenticationApi` is a context bridge to the main process, when we're running in an
+ * Electron context.
+ *
+ * # Safety
+ *
+ * We're assuming that the main process has exposed the `authenticationApi` context bridge (see
+ * `lib/client/src/preload.ts` for details), and that it contains the functions defined in this
+ * interface. Our app can't function if these assumptions are not met, so we're disabling the
+ * TypeScript checks for this interface when we use it. */
+interface AuthenticationApi {
+  /** Open a URL in the system browser. */
+  openUrlInSystemBrowser: (url: string) => void;
+  /** Set the callback to be called when the system browser redirects back to a URL in the app,
+   * via a deep link. See {@link setDeepLinkHandler} for details. */
+  setDeepLinkHandler: (callback: (url: string) => void) => void;
+}
+
```

Comment:
I've seen implementation of that in one of previous PRs. It doesn't seem to be in this file. Anyway, why we are defining an interface here for that if this thing is defined in TS sources, so we can just import its type?

Reply:
it's not exported from the module in which it's defined. as an alternative we could consider adding the interface to `env.d.ts`, and using it as a type annotation in `lib/client/src/preload.ts`
(plus augmenting `Window` in `env.d.ts`:)
```ts
interface Window {
  authenticationApi: AuthenticationApi;
}
```

Reply:
Wait, but why we just don't simply export it from where it is defined? Probably Im missing something obvious here! :)

Reply:
@wdanilo seems like it's exposed via `exposeInMainWorld` [here](https://github.com/enso-org/enso/blob/dd1144e9ab7169f47da866028bc1d84c5a1b8fbd/app/ide-desktop/lib/client/src/preload.ts#L92). also it's probably not a good idea to import things from sibling modules. these might be combined when we combine the electron and browser code though

Reply:
Ok, it's OK for me to know that you guys are OK with the solution that is here. I'm not so deeply in this code, so I'm OK as well. Let's keep it this way, but in case we could simplify it after combining the code (as you mentioned), let's do it :)